lib/storage-service.js

```diff
@@ -66,18 +66,28 @@ class StorageService {
    */
   lookup(token, secret, ProviderType) {
     return this._getLeaseManager(token, secret, ProviderType).then((manager) => {
+      const data = {
+        correlation_id: manager.correlation_id
+      };
+
       if (manager.status === STATUS.READY) {
+        StorageService.logManagerEvent('INFO', 'Manager was able to lookup a secret.', manager);
+
         // LeaseManager is ready, immediately resolve with data
-        return manager.data;
+        return Object.assign(data, manager.data);
       }
 
       // Queue the promise states with a timeout
-      return onceWithTimeout(manager, 'ready', this.timeout)
-        .then(() => manager.data)
-        .catch((err) => {
-          this._managers.delete(manager.name);
-          throw err;
-        });
+      return onceWithTimeout(manager, 'ready', this.timeout).then(() => {
+        StorageService.logManagerEvent('INFO', 'Manager was able to lookup a secret.', manager);
+
+        return Object.assign(data, manager.data);
+      })
+      .catch((err) => {
+        StorageService.logManagerEvent('ERROR', err, manager);
+        this._managers.delete(manager.name);
```

Comment:
I tested this because I had a sneaking suspicion we talked this through before. Turns out if you pass `undefined` to the `Map.delete` method it just returns `false` and doesn't throw.

The hoisting issue is fixed in 425a9f4.